src/arc_utilities/algorithms.py

```diff
@@ -38,3 +39,40 @@ def zip_repeat_shorter(*args):
     zipped = zip(*[repeat_last(x) for x in args])
     for i in range(longest_len):
         yield next(zipped)
+
+
+def chunked_iterable(iterable, size):
+    """
+
+    Args:
+        iterable: an iterable to split into chunks. could be a list, tuple, or generator
+        size: the size of the chunks. If the iterable cannot be evenly split, the final elements will be dropped.
```

Comment:
This seems like a potential source for a lot of bugs; better to have optional parameter toggling whether it should be dropped or a smaller last chunk should be returned

Reply:
If someone wants smaller last chunk, they would use the standard  chunking modules for that (in `itertools` or `more_itertools`) I made this one specifically for a case those modules do not support. Perhaps a better function name? `chunked_drop_last` ?

Reply:
Yeah a more specific function name would be good